build-in-container/scripts/setup.sh

```diff
@@ -59,6 +59,36 @@ check_specs() {
     fi
 }
 
+# setup custom repo to install RPMs from
+setup_custom_repo() {
+    echo "------------ Setting up custom repo ------------"
+    if [[ ! -z "${RPM_repo}" ]]; then
+        echo "------------ RPM_repo is $RPM_repo ------------"
+        cp $MARINER_BASE_DIR/scripts/custom-repo.repo $MARINER_BASE_DIR/toolkit/repos/
+        sed -i "s~<CUSTOM_REPO_BASE_URL>~${RPM_repo}~" $MARINER_BASE_DIR/toolkit/repos/custom-repo.repo
+        cat $MARINER_BASE_DIR/toolkit/repos/custom-repo.repo >> $MARINER_BASE_DIR/toolkit/resources/manifests/package/local.repo
+    fi
+
+    if [[ ! -z "${RPM_storage}" ]]; then
+        echo "------------ RPM_storage is $RPM_storage ------------"
+        #install azcopy
+        wget -O azcopy_v10.tar.gz https://aka.ms/downloadazcopy-v10-linux  || { echo "ERROR: Could not install azcopy"; exit 1; }
+        tar -xf azcopy_v10.tar.gz --strip-components=1
+        mv azcopy /bin/
+        rm -rf azcopy* NOTICE.txt
+
+        #download all RPMs from Azure $RPM_storage to $MARINER_BASE_DIR/build/rpm_cache/cache
+        azcopy copy $RPM_storage/* $MARINER_BASE_DIR/build/rpm_cache/cache
```

Comment:
This feels like a very niche use case, there is no requirement a user will be using an azure blob store (does PMC work with it?). We need to support the features listed here: https://github.com/microsoft/CBL-Mariner/blob/2.0/toolkit/docs/building/building.md#keys-certs-and-remote-sources (most critically `PACKAGE_URL_LIST`).